server/graphql/types/src/types/program/program_event.rs

```diff
@@ -64,18 +64,28 @@ impl ProgramEventSortInput {
 
 pub struct ProgramEventNode {
     pub store_id: String,
-    pub row: ProgramEventRow,
+    pub program_event: ProgramEvent,
     pub allowed_ctx: Vec<String>,
 }
 
+impl ProgramEventNode {
+    fn row(&self) -> &ProgramEventRow {
+        &self.program_event.program_event_row
+    }
+
+    fn name_row(&self) -> &Option<NameRow> {
+        &self.program_event.name_row
+    }
+}
+
 #[Object]
 impl ProgramEventNode {
-    pub async fn patient_id(&self) -> &Option<String> {
-        &self.row.patient_id
+    pub async fn patient_id(&self) -> Option<String> {
+        self.name_row().as_ref().map(|it| it.id.clone())
     }
 
     pub async fn patient(&self, ctx: &Context<'_>) -> Result<Option<PatientNode>> {
-        let Some(patient_id) = &self.row.patient_id else {
+        let Some(patient_id) = self.name_row().as_ref().map(|it| &it.id) else {
```

Comment:
Before it was just a `patient_id: Option<String>` and now its a `name: Option<NameRow>`. In Typescript you would do `name?.id`, i.e. check if name is Some and then get `name.id`. 

In Rust there is `Option::map()` to check if the Option is Some and if this is the case it takes the Option value and you can transform it to another value. For example, using `name.map(|it| it.id)` `Option<NameRow>` becomes `Option<String>`. You then need as_ref because you don't want to consume the NameRow in the option (because we only have a reference to the option in the first place...

Reply:
> Also might be easier to explain in person but why does `patient_link_id` implementation require doing this in the gql layer? We are generally avoiding touching anything but the repo layer and the odd test that breaks.

Not quite sure what you mean, I avoid accessing the patient_link_id by using the NameRow which has been joined in the repo layer. Yeah might be easier to discuss in person.